data_structures/src/chain/priority.rs

```diff
@@ -0,0 +1,1169 @@
+    #[default]
+    Unknown,
+    /// The time-to-block is up to X epochs.
+    UpTo(u32),
+}
+
+impl TimeToBlock {
+    /// Convert a `TimeToBlock` into seconds.
+    ///
+    /// This requires knowledge of the number of seconds per epoch, aka __checkpoint period__.
+    pub fn as_secs(&self, seconds_per_epoch: u16) -> Option<u32> {
+        match self {
+            TimeToBlock::Around(x) => Some(x * seconds_per_epoch as u32),
+            TimeToBlock::Exactly(x) => Some(x * seconds_per_epoch as u32),
+            TimeToBlock::LessThan(x) => Some(x * seconds_per_epoch as u32),
+            TimeToBlock::Unknown => None,
+            TimeToBlock::UpTo(x) => Some(x * seconds_per_epoch as u32),
+        }
+    }
+
+    /// Convert a `TimeToBlock` into a formatted string expressing the time-to-block in seconds.
+    ///
+    /// This requires knowledge of the number of seconds per epoch, aka __checkpoint period__.
+    pub fn pretty_print_secs(&self, seconds_per_epoch: u16) -> String {
+        fn unit_change(seconds: u32, divider: u32) -> (u32, u32, String) {
+            let value = seconds / divider;
+            let remainder = seconds % divider;
+            let plural = String::from(if value == 1 { "" } else { "s" });
+
+            (value, remainder, plural)
+        }
+
+        let string = self
+            .as_secs(seconds_per_epoch)
+            .map(|seconds| {
+                let mut strings = Vec::<String>::new();
+                let (days, remainder, plural) = unit_change(seconds, 24 * 60 * 60);
+                if days > 0 {
+                    strings.push(format!("{} day{}", days, plural))
+                }
+                let (hours, remainder, plural) = unit_change(remainder, 60 * 60);
+                if hours > 0 {
+                    let separator = if strings.is_empty() {
+                        ""
+                    } else if remainder > 0 {
+                        ", "
+                    } else {
+                        " and "
+                    };
+
+                    strings.push(format!("{}{} hour{}", separator, hours, plural))
+                }
+                let (minutes, remainder, plural) = unit_change(remainder, 60);
+                if minutes > 0 {
+                    let separator = if strings.is_empty() {
+                        ""
+                    } else if remainder > 0 {
+                        ", "
+                    } else {
+                        " and "
+                    };
+                    strings.push(format!("{}{} minute{}", separator, minutes, plural))
+                }
+                let (seconds, _, plural) = unit_change(remainder, 1);
+                if seconds > 0 || (minutes == 0 && hours == 0 && days == 0) {
+                    let separator = if strings.is_empty() { "" } else { " and " };
+                    strings.push(format!("{}{} second{}", separator, seconds, plural));
+                }
+
+                strings.join("")
```

Comment:
This logic to convert seconds to a human readable string does not belong here. We already have a similar function called `seconds_to_human_string` in the `witnet_utils` crate, so if using that function is not possible, I would prefer to move this logic to that crate and create a new function. And the tests should be moved there as well. Or even better, use an external dependency that handles this conversion, then we don't need to add as many tests.

Another possibility that comes to mind is to return the time in seconds from the API, and let the client handle the conversion to a human readable string.

Reply:
> Another possibility that comes to mind is to return the time in seconds from the API, and let the client handle the conversion to a human readable string.

That is roughly what is happening here. The JSON-RPC API is expressing it in epochs and the CLI is the client handling the conversion to a human readable string.

Reply:
> Another possibility that comes to mind is to return the time in seconds from the API, and let the client handle the conversion to a human readable string.

The API uses seconds now :wink: